File: dyd/roots/libraries/dhall-codegen-snapshot-tests/dyd/assets~fixture=person+target=py-3.11/expected.py
from __future__ import annotations

from collections.abc import Awaitable, Callable
from datetime import date, datetime, time
from typing import Annotated, Any, Literal, Protocol, TypeAlias

from pydantic import BaseModel, TypeAdapter

# used to simulate intersection type
def _check_all_of(*adapters: TypeAdapter[Any]):
    def check(v: Any) -> Any:
        for ta in adapters:
            ta.validate_python(v)  # raises on failure
        return v
    return check


class PersonContact(BaseModel):# contact details
    phone_number : str
    email : (str) | None

class Person(BaseModel):
    appointment_time : time
    binary_data : str
    birth_date : str
    contact : PersonContact
    created_at : str
    date_of_birth : date
    destination_ip : str
    friends : list[Person]
    id : str
    name : str
    retention_period : str
    source_ip : str
    token : str

File: dyd/roots/libraries/dhall-codegen-snapshot-tests/dyd/assets~fixture=person+target=ts/expected.ts
export type Person =
	{
		appointment_time :
			/** RFC 3339 time */
			string;
		birth_date :
			/** ISO 8601 calendar date */
			string;
		created_at :
			/** RFC 3339 date-time */
			string;
		date_of_birth :
			/** date of birth */
			Date;
		friends :
			/** friends */
			Array<
				Person
			>;
		id :
			/** RFC 4122 UUID */
			string;
		name :
			/** full name */
			string;
		retention_period :
			/** ISO 8601 duration */
			string;
		contact_email ?:
			/** contact email (we might not have this) */
			string;
	};